DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,153 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers.ExternalApis
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Models.LearningHubApiClient;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public interface ILearningHubApiClient
+    {
+        Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        );
+
+        Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        );
+    }
+
+    public class LearningHubApiClient : ILearningHubApiClient
+    {
+        private readonly HttpClient client;
+        private readonly string learningHubApiBaseUrl;
+
+        public LearningHubApiClient(HttpClient httpClient, IConfiguration config)
+        {
+            string learningHubApiKey = config.GetLearningHubApiKey();
+            learningHubApiBaseUrl =
+                "https://uks-learninghubnhsuk-openapi-dev.azurewebsites.net";
+
+            client = httpClient;
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Add("X-API-KEY", learningHubApiKey);
+        }
+
+        public async Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        )
+        {
+            const string searchEndpoint = "/Resource/Search";
+
+            var endpointPath = learningHubApiBaseUrl + searchEndpoint;
+            var requestUrl = GetUrlWithSearchQueryParams(endpointPath, text, offset, limit, resourceTypes);
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<ResourceSearchResult>(response);
+            return result;
+        }
+
+        public async Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId)
+        {
+            const string resourceEndpoint = "/Resource/";
+            var requestUrl = learningHubApiBaseUrl + resourceEndpoint + resourceReferenceId;

```

Comment:
I think we should just string interpolate this into `$"Resource/{resourceReferenceId}"`

Reply:
done